### PR TITLE
Put BN layers into evaluation mode during AutoQ search in OD sample

### PR DESCRIPTION
**Pull request by vanyalzr**
BN statistics were calculated batch-wise during evaluation in the OD sample, which technically gives incorrect metric values.

### CODE REVIEW

**Review by vshampor — APPROVED** [review]

File: examples/object_detection/main.py [hunk 1]
```diff
@@ -142,6 +142,11 @@ def criterion_fn(model_outputs, target, criterion):
             return loss_l + loss_c
 
         def autoq_test_fn(model, eval_loader):
+            # Put BN layers into evaludtion mode
```
Comment (ljaljushkin) [inline examples/object_detection/main.py:145]: why is it applicable for object detection only?
does it make sense to do it for all models inside autoq before calling autoq_test_fn?

Comment (vanyalzr) [reply examples/object_detection/main.py:145]: Will move eval mode switch for BNs inside `test_net` in this sample. For other samples, we already have `model.eval()` inside validation methods. I guess it's the responsibility of the user/training sample to do correct validation of the model.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]
After discussion with Ivan, we decided to refactor test_net of the OD sample to explicitly put BN in eval mode.
@vshampor, please don't merge

**Comment by ljaljushkin** [comment]
@vuiseng9, it might affect AutoQ results for object detection, because mAP became a slightly different.

**Review by ljaljushkin — APPROVED** [review]
